singer/src/main/java/com/pinterest/singer/utils/LogConfigUtils.java

```diff
@@ -715,30 +721,72 @@ public static LogStreamWriterConfig parseLogStreamWriterConfig(AbstractConfigura
-    case PULSAR:
-      writerConfig.setPulsarWriterConfig(parsePulsarWriterConfig(
-          new SubsetConfiguration(writerConfiguration, writerTypeString + ".")));
-      return writerConfig;
-    case MEMQ:
-      writerConfig.setMemqWriterConfig(parseMemqWriterConfig(
-          new SubsetConfiguration(writerConfiguration, writerTypeString + ".")));
-      return writerConfig;
-    default:
-      throw new ConfigurationException("Unsupported log writer type.");
+      case S3:
+        writerConfig.setS3WriterConfig(parseS3WriterConfig(
+                new SubsetConfiguration(writerConfiguration, writerTypeString + ".")));
+        return writerConfig;
+      case KAFKA08:
+      case KAFKA:
+        writerConfig.setKafkaWriterConfig(parseKafkaWriterConfig(
+            new SubsetConfiguration(writerConfiguration, writerTypeString + ".")));
+        return writerConfig;
+      case REALPIN:
+        writerConfig.setRealpinWriterConfig(parseRealpinWriterConfig(
+            new SubsetConfiguration(writerConfiguration, writerTypeString + ".")));
+        return writerConfig;
+      case NO_OP:
+        writerConfig.setNoOpWriteConfig(parseNoOpWriterConfig(
+            new SubsetConfiguration(writerConfiguration, writerTypeString + ".")));
+        return writerConfig;
+      case PULSAR:
+        writerConfig.setPulsarWriterConfig(parsePulsarWriterConfig(
+            new SubsetConfiguration(writerConfiguration, writerTypeString + ".")));
+        return writerConfig;
+      case MEMQ:
+        writerConfig.setMemqWriterConfig(parseMemqWriterConfig(
+            new SubsetConfiguration(writerConfiguration, writerTypeString + ".")));
+        return writerConfig;
+      default:
+        throw new ConfigurationException("Unsupported log writer type.");
+      }
+  }
+
+  private static S3WriterConfig parseS3WriterConfig(AbstractConfiguration writerConfiguration) throws ConfigurationException {
+    writerConfiguration.setThrowExceptionOnMissing(true);
+    S3WriterConfig config = new S3WriterConfig();
+    // Required configs
+    try {
+      config.setBucket(writerConfiguration.getString(SingerConfigDef.BUCKET));
+    } catch (Exception x) {
+      throw new ConfigurationException("S3Writer bucket is required in Writer Configuration");
+    }
+
+    try {
+      config.setKeyPrefix(writerConfiguration.getString(SingerConfigDef.KEY_PREFIX));
+    } catch (Exception x) {
+      throw new ConfigurationException("S3Writer keyPrefix is required in Writer Configuration");
     }
+
+    try {
+      config.setFileNameFormat(writerConfiguration.getString(SingerConfigDef.FILE_NAME_FORMAT));
+    } catch (Exception x) {
+      throw new ConfigurationException("S3Writer fileNameFormat is required in Writer Configuration");
+    }
+
+    // Optional configs
+    if (writerConfiguration.containsKey(SingerConfigDef.MAX_FILE_SIZE_IN_BYTES)) {
+      config.setMaxFileSizeMB(writerConfiguration.getInt(SingerConfigDef.MAX_FILE_SIZE_MB));
```

Comment:
Do we just let this and below ones throw if the value is not in integer format?

Reply:
yes I believe, it would throw here and I think this is what we do for other configurations as well:

```
public int getInt(String key) {
        Integer i = this.getInteger(key, (Integer)null);
        if (i != null) {
            return i;
        } else {
            throw new NoSuchElementException('\'' + key + "' doesn't map to an existing object");
        }
    }
```